service/history/timerQueueProcessorBase.go

```diff
@@ -704,15 +705,20 @@ func (t *timerQueueProcessorBase) deleteWorkflowHistory(task *persistence.TimerT
 				tag.TaskType(task.GetTaskType()))
 			versionHistories := msBuilder.GetAllBranches()
 
-			for _, versionHistory := range versionHistories.Histories {
+		CleanupLoop:
+			for idx, versionHistory := range versionHistories.Histories {
```

Comment:
probably also need to do a null check here

Reply:
Done